skimage/measure/_regionprops.py

```diff
@@ -166,6 +228,45 @@ def __init__(self, slice, label, label_image, intensity_image,
         self._cache = {}
         self._ndim = label_image.ndim
 
+        self._extra_properties = {}
+        if extra_properties is None:
+            extra_properties = []
+        for func in extra_properties:
+            name = func.__name__
+            if hasattr(self, name):
+                msg = (
+                    f"Extra property '{name}' is shadowed by existing "
+                    "property and will be inaccessible. Consider renaming it."
+                )
+                warn(msg)
```

Comment:
Can't we dynamically edit `COL_DTYPES` to avoid the handling of  dtype for extra properties in `_props_to_dict`?
```suggestion
                warn(msg)
            else:
                COL_DTYPES[func.__name__] = _infer_regionprop_dtype(
                    func,
                    intensity=intensity_image is not None,
                    ndim=self._ndim,
                )
```

Reply:
@rfezzani  here we chose to preserve existing behaviour rather than try to improve it, which could have unintended consequences. I suggest leaving dynamic inference of existing properties for a future PR.

Reply:
@jni was faster than me. In general, I don't like modifying a global (in the namespace) state. That state change will persist in `COL_DTYPES` potentially leading to the "unintended consequences" that @jni mentioned.

Reply:
The suggestion is not an improvement to existing behavior but the modification of the extra properties (new feature) dtype handeling ^^. Avoiding namespace variable modification is a good point, no problem ;)